Dropdown: add focusOnMount prop to pass onto Popover component

psealock opened the pull request:
## Description
I'd like to use the functionality of the `<Dropdown />` component, but not bring focus to the dropdown as it gets mounted by making use of `<Popover />` prop `focusOnMount`.

This PR adds a `focusOnMount` prop to `<Dropdown />` and simple passes it down to `<Popover />`.

## How has this been tested?
I've tested via UI implementation, https://github.com/woocommerce/wc-admin/pull/1069. Changes in this PR will not affect other areas of the codebase.

## Screenshots <!-- if applicable -->

![screen shot 2018-12-12 at 4 36 44 pm](https://user-images.githubusercontent.com/1922453/49845600-34a45080-fe2c-11e8-8828-7b8b488e6106.png)

## Use case involving keyboard Interactions

The option to have the popover NOT take focus on mounting is useful in the case of the `<Dropdown />` button being an input. The contents of the `<Popover />` might be for information purposes, or in this case, provide additional methods of date selection beyond an input. 

As seen below, tabbing to the input opens the popover, but by not stealing focus, allows keyboard-only users to remain in the input with focus.

![datepicker](https://user-images.githubusercontent.com/1922453/50059864-46517380-01f1-11e9-9145-e8366bdb6033.gif)

## Types of changes
<!-- What types of changes does your code introduce?  -->
<!-- Bug fix (non-breaking change which fixes an issue) -->
<!-- New feature (non-breaking change which adds functionality) -->
<!-- Breaking change (fix or feature that would cause existing functionality to not work as expected) -->
New feature: A non-breaking change that exposes `<Popover />`'s functionality.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->
- [x] I've included developer documentation if appropriate. <!-- Handbook: https://wordpress.org/gutenberg/handbook/designers-developers/ -->

psealock committed:
Dropdown: add focusOnMount prop to pass onto Popover component

tofumatt reviewed (changes requested):
My first thought is that the example provided seems a bit confusing; the popover is displayed on... focus of the input but the keyboard isn't moved to it. I guess I'm a bit lost as to how that works for keyboard users having not tested it and just having glanced at the code.

But I'm fine with exposing this property. I think you'll want to update the package's CHANGELOG though as this is an API change and we'll need to publish a new version of it. Our packages are handled via lerna and there are old PRs you can reference for this, but let me know if the package updates aren't clear 😅

psealock commented:
Thanks for the review @tofumatt !

>I guess I'm a bit lost as to how that works for keyboard users having not tested it and just having glanced at the code.

I added a gif and explanation to illustrate how it works for keyboard users. By avoiding bringing focus to the popover contents, keyboard users aren't taken out of a form's sequence and can use the input as intended.

I also added to the CHANGELOG with a minor bump (new backwards-compatible feature) `7.1.0 (Unreleased)`.

tofumatt reviewed (approved):
Ah okay, I getcha. Looks good then; it would probably be good to have an end-to-end test to verify tabbing works as expected here, but if you want to do that in a follow-up PR that's okay with me 😄

(You'll need to fix the conflict in the `CHANGELOG` but then it's good to merge, sorry I didn't get to this for forever, I think I missed it in my issue queue. My bad.)

youknowriad commented:
Would you mind rebasing this?

psealock commented:
Thanks @youknowriad ! Another rebase did the trick, thanks for taking a look.
